File: interviewPreparationKit/dictionariesAndHashmaps/twoStrings.js
function twoStrings(s1, s2) {
    // Write your code here
    var set1 = new Set();
    var set2 = new Set();
    for (let i = 0; i < s1.length; i++) {
        set1.add(s1[i]);
    }
    for (let i = 0; i < s2.length; i++) {
        set2.add(s2[i]);
    }
    var intersection = new Set(
        [...set1].filter(x => set2.has(x))
    );
    return intersection.size > 0 ? 'YES' : 'NO';
}

File: interviewPreparationKit/dictionariesAndHashmaps/two_strings.py
def twoStrings(s1, s2):
    # Write your code here
        s1_substrings = set(s1)
        s2_substrings = set(s2)
        
        common_substrings = s1_substrings.intersection(s2_substrings)
        if common_substrings:
            return 'YES'
        else:
            return 'NO'
        # O(n) space, O(n) time